pkg/cmd/inspect.go

```diff
@@ -155,6 +155,93 @@ func (o *InspectOptions) Validate() error {
 	return nil
 }
 
+func objectReferenceToResourceInfo(clientGetter genericclioptions.RESTClientGetter, ref *configv1.ObjectReference) (*resource.Info, error) {
+	b := resource.NewBuilder(clientGetter).
+		Unstructured().
+		ResourceTypeOrNameArgs(false, fmt.Sprintf("%s/%s", ref.Resource, ref.Name)).
+		NamespaceParam(ref.Namespace).
+		Flatten().
+		Latest()
+
+	infos, err := b.Do().Infos()
+	if err != nil {
+		return nil, err
+	}
+
+	return infos[0], nil
+}
+
+func infoToContextKey(info *resource.Info) string {
+	return fmt.Sprintf("%s/%s.%s/%s/%s", info.Namespace, info.ResourceMapping().GroupVersionKind.Group, info.ResourceVersion, info.ResourceMapping().Resource.Resource, info.Name)
+}
+
+// collectPolymorphicResourceInfo receives an object to gather debugging data for, and a context to keep track of
+// already-seen objects when following related-object reference chains.
+func (o *InspectOptions) collectPolymorphicResourceInfo(info *resource.Info, context map[string]bool) ([]string, error) {
+	if seen := context[infoToContextKey(info)]; seen {
+		return nil, nil
+	}
+	context[infoToContextKey(info)] = true
+
+	unstr, ok := info.Object.(*unstructured.Unstructured)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type. Expecting %q but got %T", "*unstructured.Unstructured", info.Object)
+	}
+
+	skippedNamespaces := []string{}
+
+	switch info.ResourceMapping().Resource.Resource {
+	case "clusteroperators":
+		// save clusteroperator resources to disk
+		if err := o.gatherClusterOperatorResource(path.Join(o.baseDir, "/clusteroperator"), unstr); err != nil {
+			return nil, err
+		}
+
+		// obtain associated objects for the current clusteroperator resources
+		relatedObjReferences, err := obtainClusterOperatorRelatedObjects(unstr)
+		if err != nil {
+			return nil, err
+		}
+
+		errs := []error{}
+		for _, relatedRef := range relatedObjReferences {
+			relatedInfo, err := objectReferenceToResourceInfo(o.configFlags, relatedRef)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			skippedNs, err := o.collectPolymorphicResourceInfo(relatedInfo, context)
+			skippedNamespaces = append(skippedNamespaces, skippedNs...)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+		}
+
+		if len(errs) > 0 {
+			return skippedNamespaces, errors.NewAggregate(errs)
+		}
+	case "namespaces":
+		if err := o.gatherNamespaceData(path.Join(o.baseDir, "namespaces", info.Name), info.Name); err != nil {
+			if kapierrs.IsNotFound(err) {
+				skippedNamespaces = append(skippedNamespaces, info.Name)
+				return skippedNamespaces, err
```

Comment:
why?  We've already visited it, so there's no reason for a special case, right?

Reply:
This is for cases where the namespace is not found at all, so that we can list all of them (if any) at the end